test/llvm-intrinsics/bitreverse_small_type.ll

```diff
@@ -0,0 +1,38 @@
+;; Check that llvm.bitreverse.* intrinsics are lowered to emulation functions for
+;; 2/4-bit types.
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv -spirv-ext=+SPV_INTEL_arbitrary_precision_integers -spirv-text %t.bc -o - | FileCheck %s --check-prefix=CHECK-SPIRV
```

Comment:
That is not a vanilla SPIR-V. In which float i2 and i4 were generated?

Reply:
llvm.bitreverse.i2/4 can be generated from:

> using uint2_t = unsigned _BitInt(2);
> using uint4_t = unsigned _BitInt(4);
> uint4_t reverse4(uint4_t a) {
>     a = ((0x5 & a) << 1) | (0x5 & (a >> 1));
>     return (a << 2) | (a >> 2);
> }
> uint2_t reverse2(uint2_t a) {
>     return (a << 1) | (a >> 1);
> }

Reply:
will it be reasonable to add this code snippet as a comment here? Thanks

Reply:
Added comment with C source that generates 2/4-bit bitreverse.